### PR TITLE
Add duration to notes

### PR DESCRIPTION
**Pull request by rlwprog**
Added duration to the notes by making an extra parameter for note initialization to represent duration. There were a few different ways I saw to do this, but in order to keep with the "note.duration" format, I opted to do what I did and then use the Mary_Had_a_Little_Lamb class to make distinct notes for each duration/type and initialize them directly in the _sequence method

### CODE REVIEW

**Comment by gbroques** [comment]
Looks like you have Briton's `light_library` branch mixed in with yours.

Forgot to mention, when your create a branch, make sure your on `master` because creating a branch will copy all of the commits from whatever branch your currently on to the new branch.

I can show you how to "drop" commits tomorrow.

File: song_match/song/songs/mary_had_a_little_lamb.py [hunk 1]
```diff
@@ -10,26 +10,34 @@
 from song_match.song import Song
 from song_match.song.note import Note
 
+EIGHTH_NOTE = .25 # time for eighth note
+QUARTER_NOTE = .5 # time for quarter note
+HALF_NOTE = 1 # time half note
+WHOLE_NOTE = 2 # time for whole note
```
Comment (gbroques) [inline song_match/song/songs/mary_had_a_little_lamb.py]: 1. I would move all these constants to above the `Note` class in `note.py`.

2. Also, if you add a colon `:` after the pound sign `#`, then Sphinx will generate documentation for these.

For example, `QUARTER_NOTE = .5  #: Time for quarter note`

File: song_match/song/note.py [hunk 2]
```diff
@@ -12,10 +12,11 @@
 class Note:
     __is_mixer_initialized = False
 
-    def __init__(self, note: str):
+    def __init__(self, note: str, duration: int):
```
Comment (gbroques) [inline song_match/song/note.py]: You could default the duration to a quarter note here, that way you don't always pass in a duration when constructing `Note` instances.

File: song_match/song/songs/mary_had_a_little_lamb.py [hunk 3]
```diff
@@ -10,26 +10,34 @@
...
 
 class MaryHadALittleLamb(Song):
 
     @property
     def _notes(self) -> List[Note]:
         return [
-            Note('C4'),
-            Note('D4'),
-            Note('E4')
+            Note('C4', HALF_NOTE),
```
Comment (gbroques) [inline song_match/song/songs/mary_had_a_little_lamb.py]: You wouldn't have to pass in `HALF_NOTE` here if the duration defaulted to some value.

File: song_match/song/songs/mary_had_a_little_lamb.py (hunk 1 above)
Comment (rlwprog) [reply song_match/song/songs/mary_had_a_little_lamb.py]: Thanks for the tip for Sphinx. When I remove these constant definitions from the MHALL class and put them in the note, I can't use them while passing them to the note constructors, i.e.             Note('E4', HALF_NOTE) without it throwing an error

File: song_match/song/note.py (hunk 2 above)
Comment (rlwprog) [reply song_match/song/note.py]: Ok cook, went ahead and did that. Thanks

File: song_match/song/songs/mary_had_a_little_lamb.py (hunk 3 above)
Comment (rlwprog) [reply song_match/song/songs/mary_had_a_little_lamb.py]: Done

File: song_match/song/songs/mary_had_a_little_lamb.py (hunk 1 above)
Comment (gbroques) [reply song_match/song/songs/mary_had_a_little_lamb.py]: Did you import them?

After you move them to `note.py`, adding `from song_match.song.note import QUARTER_NOTE` to the top of the file should fix your error.

Comment (rlwprog) [reply song_match/song/songs/mary_had_a_little_lamb.py]: Oh ok that worked. Thank you!

**Pull request merged**